sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -63,4 +66,28 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
                      Function<String, Mono<PagedResponse<T>>> nextPageRetriever) {
         super(firstPageRetriever, nextPageRetriever);
     }
+
+    /**
+     * Maps this PagedFlux instance of T to a PagedFlux instance of type S as per the provided mapper function.
+     *
+     * @param mapper The mapper function to convert from type T to type S.
+     * @param <S> The mapped type.
+     * @return A PagedFlux of type S.
+     */
+    public <S> PagedFlux<S> mapTo(Function<T, S> mapper) {
```

Comment:
Can't override `map` method in Flux as the return type is not `Flux`. Open to suggestions for a better method name.

Reply:
I'm not sure what a better name is. I wonder if somehow 'page' should be in it, because we will possibly need to add more page-specific functionality in the future. We already have `byPage`, maybe this should be `mapPage`. In the future we might have `zipPage` or `transformPage` if necessary.